### PR TITLE
Use the polling crate in xclock_utc example

### PR DESCRIPTION
**Pull request by notgull**
Using an established ecosystem crate for polling is probably better than using system calls.

### CODE REVIEW

File: x11rb/Cargo.toml [hunk 1]
```diff
@@ -58,37 +58,7 @@ resource_manager = ["x11rb-protocol/resource_manager"]
...
-    "xinerama",
-    "xinput",
-    "xkb",
-    "xprint",
-    "xselinux",
-    "xtest",
-    "xv",
-    "xvmc"
-]
+all-extensions = ["x11rb-protocol/all-extensions", "composite", "damage", "dbe", "dpms", "dri2", "dri3", "glx", "present", "randr", "record", "render", "res", "screensaver", "shape", "shm", "sync", "xevie", "xf86dri", "xf86vidmode", "xfixes", "xinerama", "xinput", "xkb", "xprint", "xselinux", "xtest", "xv", "xvmc"]
```
Comment (psychon) [inline x11rb/Cargo.toml:61]: Huh? I liked the previous version more...

**Review by psychon — COMMENTED** [review]
Thanks. This seems convincing. "Going epoll" might be a bit too much for this, but getting rid of `#[cfg]` is definitely nice.

While looking at the polling crate, I came across their use of the `log` (and an issue to switch to `tracing`). What do you think: Would it make sense to add one of these as an (optional) dependency to x11rb?

**Comment by psychon** [comment]
For future me: There was a successful AppVeyor build of the current state of things.

File: x11rb/Cargo.toml (hunk 1 above)
Comment (notgull) [reply x11rb/Cargo.toml:61]: 🙈 I keep forgetting I can't use `cargo add` on this crate. Oopsie daisy

**Review by psychon — APPROVED** [review]

**Comment by notgull** [comment]
> While looking at the polling crate, I came across their use of the `log` (and an issue to switch to `tracing`). What do you think: Would it make sense to add one of these as an (optional) dependency to x11rb?

The only reason we aren't using `tracing` is because tracing has an MSRV of 1.56 and the MSRV for `polling` is currently limited to the current Debian Stable (1.48, hopefully being bumped soon, see smol-rs/smol#244). I would support adding `tracing` logging to this crate.

**Comment by psychon** [comment]
Okay, thanks. I'm not quite sure about `log` vs `tracing`, but they seem interoperable in both directions...

> I would support adding tracing logging to this crate.

Also if I were to make it an optional dependency and hack together an abstraction to make that possible? :see_no_evil:

**Pull request merged**